File: Next_Layer_Boost_App-main/app/src/main/java/com/sis/clighteningboost/Utills/SocketInstance.java
package com.sis.clighteningboost.Utills;

import android.app.Application;

import java.net.URISyntaxException;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import io.socket.client.IO;
import io.socket.client.Socket;

public class SocketInstance extends Application {
    String mAccessToken;
    public SocketInstance(String accessToken){
        this.mAccessToken = accessToken;
    }
    private Socket mSocket;
    {
        IO.Options options = new IO.Options();
        Map<String, List<String>> headers = new HashMap<>();
        String bearer = "Bearer "+ mAccessToken;
        headers.put("Authorization", Arrays.asList(bearer));
        options.extraHeaders = headers;

        try {
            mSocket = IO.socket("https://realtime.nextlayer.live",options);
        } catch (URISyntaxException e) {
            throw new RuntimeException(e);
        }
    }
    public Socket getSocket() {
        return mSocket;
    }
}

File: Next_Layer_Boost_App-main/app/src/main/java/com/sis/clighteningboost/utils/SocketInstance.kt
package com.sis.clighteningboost.utils

import android.app.Application
import io.socket.client.IO
import io.socket.client.Socket
import java.lang.RuntimeException
import java.net.URISyntaxException
import java.util.*

class SocketInstance(accessToken: String) : Application() {

    var socket: Socket? = null

    init {
        val options = IO.Options()
        val headers: MutableMap<String, List<String>> = HashMap()
        val bearer = "Bearer $accessToken"
        headers["Authorization"] = Arrays.asList(bearer)
        options.extraHeaders = headers
        try {
            socket = IO.socket("https://realtime.nextlayer.live", options)
        } catch (e: URISyntaxException) {
            throw RuntimeException(e)
        }
    }


}